File: university-study/算法/2022软院码拉松/code/找等差数列/C/main.c
#include<stdio.h>
int longestSubsequence(int* arr,int length, int difference) {
    if (length == 1) {
        return 0;
    }
    int count = 1;
    int ans = 0;
    for (int i = 0; i < length - 1; i++) {
        int temp = arr[i];
        for (int j = i+1; j < length; j++) {
            if(arr[j] == temp + difference){
                count++;
                temp = arr[j];
            }
        }
        ans = ans>count?ans:count;
        count = 1;
    }

    return ans;
}

int main(){
    int n;
    scanf("%d",&n);
    int length = n;
    int arr[n];
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }
    int difference;
    scanf("%d",&difference);
    printf("%d",longestSubsequence(arr,length,difference));
}

File: university-study/算法/2022软院码拉松/code/找等差数列/Main/src/Main.java
import java.util.Scanner;

class Main {
    public static int longestSubsequence(int[] arr, int difference) {
        if (arr.length == 1) {
            return 0;
        }
        int count = 1;
        int ans = 0;
        for (int i = 0; i < arr.length - 1; i++) {
            int temp = arr[i];
            for (int j = i+1; j < arr.length; j++) {
                if(arr[j] == temp + difference){
                    count++;
                    temp = arr[j];
                }
            }
            ans = Math.max(ans,count);
            count = 1;
        }

        return ans;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }
        int difference = scanner.nextInt();
        System.out.println(longestSubsequence(arr,difference));
    }
}